src/utils/tokens.ts

```diff
@@ -18,31 +20,74 @@ import {
   ThreeRouteFa12Token,
   ThreeRouteFa2Token,
   getChains,
-  THREE_ROUTE_SIRS_SYMBOL
+  THREE_ROUTE_SIRS_SYMBOL,
+  ThreeRouteSwapResponse,
+  ThreeRouteDex
 } from './three-route';
 import { BcdTokenData, mapTzktTokenDataToBcdTokenData, tokensMetadataProvider } from './tzkt';
 
+interface SwapsResponse {
+  directSwap: ThreeRouteSwapResponse;
+  invertedSwap: ThreeRouteSwapResponse;
+  updatedAt: string;
+}
+
+interface TokenExchangeRateEntry {
+  tokenAddress: string;
+  tokenId?: number;
+  exchangeRate: BigNumber;
+  metadata?: BcdTokenData;
+  swapsUpdatedAt?: string;
+}
+
+class TimeoutError extends Error {}
+
 const tokensListProvider = new SingleQueryDataProvider(30000, () => getThreeRouteTokens({}));
 const dexesListProvider = new SingleQueryDataProvider(30000, () => getThreeRouteDexes({}));
 
 const THREE_ROUTE_TEZ_SYMBOL = 'XTZ';
-const PROBE_TEZ_AMOUNT = 10;
-const probeSwapsProvider = new DataProvider(Infinity, async (outputTokenSymbol: string) => {
+const EMPTY_SWAP = {
+  input: 0,
+  output: 0,
+  chains: []
+};
+
+const ASPENCOIN_ADDRESS = 'KT1S5iPRQ612wcNm6mXDqDhTNegGFcvTV7vM';
```

Comment:
I think it's safe to remove this custom logic for this coin – it doesn't seem to be used or transferred at all

Reply:
You're right, the last transfer was on January 31st